panels/config/frontend/demo/ha-card-chooser-z.html

```diff
@@ -0,0 +1,43 @@
+<link rel="import" href="./new/ha-group-card.html">
+<link rel="import" href="./new/ha-state-card.html">
+
+<script>
+class HaCardChooserZ extends Polymer.Element {
+  static get is() { return 'ha-card-chooser-z'; }
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+        observer: 'hassChanged'
+      },
+      cardItem: {
+        type: Object,
+        observer: 'cardItemChanged'
+      },
+      fullCard: Boolean // todo
+    };
+  }
+
+  hassChanged(hass) {
+    const cardRoot = this;
+
+    for(let i = 0; i < cardRoot.children.length; i++) {
+      cardRoot.children[i].hass = hass;
+    }
+  }
+
+  cardItemChanged(cardItem) {
+    const cardRoot = this;
+    while (cardRoot.lastChild) {
+      cardRoot.removeChild(cardRoot.lastChild);
+    }
+    const elementTag = cardItem && cardItem.config && cardItem.config.card ||
+      cardItem.type === 'group' ? 'ha-group-card' : 'ha-state-card';
+    const element = document.createElement(elementTag);
+    element.cardItem = cardItem;
+    element.hass = this.hass;
+    cardRoot.appendChild(element);
```

Comment:
We should do all this logic of this function in the parent. We don't need a card chooser. That is just a function that returns a tag name. That's all the card choosing we need. As you see in my example in my comment, it's the overview that renders all cards at once. Not a webcomponent rendering a single card.

Reply:
ok